src/dbus_api/api.rs

```diff
@@ -105,6 +105,23 @@ fn destroy_pool(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 
     let msg = match dbus_context.engine.borrow_mut().destroy_pool(pool_uuid) {
         Ok(action) => {
+            let bd_iface_name = format!("{}.{}", STRATIS_BASE_SERVICE, "blockdev");
```

Comment:
I would generalize this in a number of ways.

* Forget about the interface. That's putting engine brains into the D-Bus API. Just search on the parent path. Note that everything has an OPContext, so you don't have to worry about what type it is.
* Consider searching recursively, just to be sure that you remove everything.
* Abstract this operation, of removing all things that have a parent that is the thing that you're removing into a separate function.
* Consider wrapping the ```push_remove``` action in some macro or something, such that the clean up of dependent objects happens automatically.